l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,478 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {EfficientCall} from "@matterlabs/zksync-contracts/l2/system-contracts/libraries/EfficientCall.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @title ConsensusRegistry
+/// @dev Manages consensus nodes and committees for the L2 consensus protocol,
+/// owned by a governance protocol. Nodes act as both validators and attesters,
+/// each playing a distinct role in the consensus process. This contract facilitates
+/// the rotation of validator and attester committees, which represent a subset of nodes
+/// expected to actively participate in the consensus process during a specific time window.
+contract ConsensusRegistry is Ownable2Step {
```

Comment:
Does it make sense to keep this contract upgradable? This we can change its implementation on flight without need to change server config.

Reply:
@brunoffranca thought to keep it as is for PoA. Upgradability will be needed for PoS, and we'll have a different version for that.

Reply:
How do you want to make upgrades on the server then? I would rather make it upgradable from day 0 so we can upgrade functionality (or fix bugs) without stoping the node